hack/build-flags.sh

```diff
@@ -28,29 +26,13 @@ function build_flags() {
     [[ -n "${commit}" ]] || abort "error getting the current commit"
     version="v$(date +%Y%m%d)-local-${commit}"
   fi
-  # For Eventing and Serving versionings,
-  # major and minor versions are the same as client version
-  # patch version is from each technical version
-  technical_version_serving=$(grep "knative.dev/serving " "${base}/go.mod" \
-    | sed -s 's/.* \(v.[\.0-9]*\).*/\1/')
-  technical_version_eventing=$(grep "knative.dev/eventing " "${base}/go.mod" \
-    | sed -s 's/.* \(v.[\.0-9]*\).*/\1/')
-  local version_serving version_eventing
-  if [[ -n "${major_minor}" ]]; then
-    version_serving=${major_minor}.$(echo ${technical_version_serving} |cut -f3 -d.)
-    version_eventing=${major_minor}.$(echo ${technical_version_eventing} |cut -f3 -d.)
-  else
-    version_serving="${technical_version_serving}"
-    version_eventing="${technical_version_eventing}"
-  fi
-  # Export as env variables to be used in `ko` OCI image build
   export KN_BUILD_VERSION="${version}"
   export KN_BUILD_DATE="${now}"
   export KN_BUILD_GITREV="${rev}"
-  echo "-X '${pkg}.BuildDate=${now}' \
+  export KN_BUILD_LD_FLAGS="-X '${pkg}.BuildDate=${now}' \
```

Comment:
I think you don't even need `export` here as you are using the variables directly and do not need to propagate them to a subprocess.